sign/sign.go

```diff
@@ -66,12 +70,56 @@ func (s *Signer) UploadBlob(path string) error {
 func (s *Signer) SignImage(reference string) (*SignedObject, error) {
 	s.log.Infof("Signing reference: %s", reference)
 
-	// TODO: unimplemented
+	if s.options.Experimental {
+		os.Setenv("COSIGN_EXPERIMENTAL", "true")
+	}
+
+	ko := sign.KeyOpts{
+		KeyRef:     s.options.KeyPath,
+		PassFunc:   generate.GetPass,
+		FulcioURL:  s.options.FulcioURL,
+		RekorURL:   s.options.RekorURL,
+		OIDCIssuer: s.options.OIDCIssuer,
+
+		InsecureSkipFulcioVerify: false,
+	}
+	regOpts := cliOpts.RegistryOptions{
+		AllowInsecure: true,
+	}
+
+	imgs := []string{reference}
+
+	outputSignature := ""
+	if s.options.OutputSignaturePath == "" {
+		outputSignature = s.options.OutputSignaturePath
+	}
+
+	outputCertificate := ""
+	if s.options.OutputCertificatePath == "" {
+		outputCertificate = s.options.OutputCertificatePath
+	}
+
+	upload := s.options.UploadToTLOG
+	force := s.options.Force
+
+	// not used right now
+	certPath := ""
+	payloadPath := ""
+	recursive := false
+	attachment := ""
```

Comment:
No need to have them if they're not used.

Reply:
ok, thanks for your review

Reply:
removed